readthedocs/search/faceted_search.py

```diff
@@ -96,53 +95,112 @@ class PageSearchBase(RTDFacetedSearch):
     }
     doc_types = [PageDocument]
     index = PageDocument._doc_type.index
-    fields = ['title^10', 'headers^5', 'content']
+
+    outer_fields = ['title^10']
+    section_fields = ['sections.title^5', 'sections.content']
+    domain_fields = [
+        'domains.type_display',
+        'domains.doc_display',
+        'domains.name',
+        'domains.display_name',
+    ]
+    fields = outer_fields
+
+    # need to search for both 'and' and 'or' operations
+    # the score of and should be higher as it satisfies both or and and
     operators = ['and', 'or']
 
+    def query(self, search, query):
+        """Manipulates query to support nested query."""
+        search = search.highlight_options(encoder='html', number_of_fragments=3)
```

Comment:
Is the `number_of_fragments` should be **3**.
I think it should be **1** now.

Reply:
Interesting, is this used in the subqueries, or just the main query?

Reply:
I don't think that it is being used in subqueries.